pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -50,6 +50,12 @@ type defaultStatefulSetControl struct {
 	podControl StatefulPodControlInterface
 }
 
+// UpdateStatefulSet executes the core logic loop for a stateful set, applying the predictable and
+// consistent monotonic update strategy by default - scale up proceeds in ordinal order, no new pod
+// is created while any pod is unhealthy, and pods are terminated in descending order. The burst
+// strategy allows these constraints to be relaxed - pods will be created eagerly and terminated
+// in bulk. Clients using the burst strategy should be careful to ensure they understand the consistency
+// implications of having unpredictable numbers of pods available.
 func (ssc *defaultStatefulSetControl) UpdateStatefulSet(set *apps.StatefulSet, pods []*v1.Pod) error {
```

Comment:
Update method internal comments to be consistent with the behavior of out of order creation and termination.